Type 'Observable<{}>' is not assignable to type 'Observable<Action>'.

## Issue Name

### Summary
dataset save effect is missing action in output


### Steps to Reproduce
See build in jenkins
https://jenkins.esss.dk/dm/job/scicat/job/catanie/job/develop/304/console

### Current Behaviour

ERROR in src/app/state-management/effects/datasets.effects.ts(43,13): error TS2322: Type 'Observable<{}>' is not assignable to type 'Observable<Action>'.
  Type '{}' is not assignable to type 'Action'.
    Property 'type' is missing in type '{}'.
[39m[22m[0m[39m[22m
[91m10 02 2019 16:00:48.936:ERROR [karma-server]: [39m{ Error: spawn ps ENOENT
    at _errnoException (util.js:1022:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn ps',
  path: 'ps',
### Expected Behaviour
builds correctly 

### Extra Details

Here you should include details about the system (if it is unique) and possible information about a fix (feel free to link to code where relevant). Screenshots/GIFs are also fine here.

--- a/src/app/state-management/actions/datasets.actions.ts
+++ b/src/app/state-management/actions/datasets.actions.ts
@@ -336,7 +336,7 @@ export class SaveDatasetAction implements Action {
 export class SaveDatasetCompleteAction implements Action {
   readonly type = SAVE_DATASET_COMPLETE;
 
-  constructor(readonly datasets: Dataset[]) {
+  constructor(readonly datasets: Dataset) {
   }
 }
 

--- a/src/app/state-management/effects/datasets.effects.ts
+++ b/src/app/state-management/effects/datasets.effects.ts
@@ -44,7 +44,10 @@ export class DatasetEffects {
     ofType(DatasetActions.SAVE_DATASET),
     map((action: DatasetActions.SaveDatasetAction) => action.dataset),
     switchMap(dataset => {
-      return this.datasetApi.upsert(dataset);
+      return this.datasetApi.upsert(dataset).pipe(
+        map(res => new DatasetActions.SaveDatasetCompleteAction(dataset)),
+        catchError(err => of(new DatasetActions.SaveDatasetFailedAction()))
+      );
     })
   );
 

